java/src/org/openqa/selenium/grid/sessionqueue/local/LocalNewSessionQueue.java

```diff
@@ -233,15 +230,6 @@ public HttpResponse addToQueue(SessionRequest request) {
                 new SessionNotCreatedException("An error occurred creating the session", e));
       }
 
-      Lock writeLock = this.lock.writeLock();
-      writeLock.lock();
-      try {
-        requests.remove(request.getRequestId());
-        queue.remove(request);
-      } finally {
-        writeLock.unlock();
-      }
-
```

Comment:
I removed it because, on success, or failure, the "complete" method will be called (or the timeoutSessions), and so, queue and requests will be cleaned.
Moreover, here, it seems `contexts.remove(reqId)` is missing

Reply:
The `contexts.remove(reqId)` is intentionally missing here to allow `retryAddToQueue` reading it.
If it is safe to remove the other code i am not sure, someone else has to review.

Reply:
ok, you are the expert :)